Slice bounds out of range in idn.decode

```go
package main

import (
        "github.com/miekg/dns/idn"
)

func main() {
        idn.FromPunycode("xn--a000000000")
}
```

```
panic: runtime error: slice bounds out of range

goroutine 1 [running]:
github.com/miekg/dns/idn.decode(0xc820069ed1, 0x0, 0xf, 0x0, 0x0, 0x0)
        /home/erik/src/github.com/miekg/dns/idn/punycode.go:271 +0x80a
github.com/miekg/dns/idn.FromPunycode(0x69f130, 0x11, 0x0, 0x0)
        /home/erik/src/github.com/miekg/dns/idn/punycode.go:65 +0x1ce
main.main()
        /home/erik/example.go:8 +0x2c

goroutine 17 [syscall, locked to thread]:
runtime.goexit()
        /home/erik/go1.5rc1/src/runtime/asm_amd64.s:1696 +0x1
exit status 2
```

The exact same for `xn--a000000000`

--- a/idn/punycode.go
+++ b/idn/punycode.go
@@ -254,6 +254,10 @@ func decode(b []byte) []byte {
 				return src
 			}
 			i += digit * w
+			if i < 0 {
+				// safety check for rune overflow
+				return src
+			}
 
 			t = tfunc(k, bias)
 			if digit < t {

--- a/idn/punycode_test.go
+++ b/idn/punycode_test.go
@@ -86,7 +86,7 @@ var invalidACEs = []string{
 	"xn--*",
 	"xn--",
 	"xn---",
-	//"xn--a000000000", // crashes IDN. TODO(miek): Fix.
+	"xn--a000000000",
 }
 
 func TestInvalidPunycode(t *testing.T) {

